File: ta_mobile/lib/models/pasien_model.dart
class Pasien {
  final int id;
  final String nama;
  final String tglLahir;
  final String gender;
  final String noTelp;
  final String alamat;
  final String namaDokter;

  Pasien({
    required this.id,
    required this.nama,
    required this.tglLahir,
    required this.gender,
    required this.noTelp,
    required this.alamat,
    required this.namaDokter,
  });

  factory Pasien.fromJson(Map<String, dynamic> json) {
    return Pasien(
      id: json['id'],
      nama: json['nama'],
      tglLahir: json['tgl_lahir'],
      gender: json['gender'],
      noTelp: json['no_telp'],
      alamat: json['alamat'],
      namaDokter: json['dokter']?['nama_dokter'] ?? '-',
    );
  }
}

File: backend/models/PasienModel.js
import { Sequelize } from "sequelize";
import db from "../config/Database.js";
import Dokter from "./DokterModel.js";

// Membuat tabel "pasien"
const Pasien = db.define("pasien", {
  id: {
    type: Sequelize.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  nama: {
    type: Sequelize.STRING,
    allowNull: false,
  },
  tgl_lahir: {
    type: Sequelize.DATEONLY,
    allowNull: false,
  },
  gender: {
    type: Sequelize.STRING,
    allowNull: false,
  },
  no_telp: {
    type: Sequelize.STRING,
    allowNull: false,
  },
  alamat: {
    type: Sequelize.STRING,
    allowNull: false,
  },
  id_dokter: {
    type: Sequelize.INTEGER,
    allowNull: false,
    references: {
      model: Dokter,
      key: "id_dokter",
    },
  },
}, {
  tableName: "pasien",
  freezeTableName: true,
  timestamps: false,
});

Pasien.belongsTo(Dokter, { foreignKey: "id_dokter" });

db.sync().then(() => console.log("Database synced"));

export default Pasien;